### PR TITLE
Implement audioProcessorChanged and latency APIs

### PR DESCRIPTION
**Pull request by jatinchowdhury18**
I was trying to hack together a couple fixes for things that weren't working in the CLAP version of BYOD. I'm not sure if these are implemented correctly, but they seem to be working in BYOD so far!

### CODE REVIEW

File: src/wrapper/clap-juce-wrapper.cpp [hunk 1]
```diff
@@ -517,6 +536,19 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
...
     }
 
+    bool implementsLatency() const noexcept override { return true; }
+    uint32_t latencyGet() const noexcept override
+    {
+        return (uint32_t)processor->getLatencySamples();
+    }
+
+    bool implementsTail() const noexcept override { return true; }
+    uint32_t tailGet(const clap_plugin_t *) const noexcept override
```
Comment (jatinchowdhury18) [inline src/wrapper/clap-juce-wrapper.cpp:546]: I didn't see this in the documentation, but I'm guessing the tail should be reported in samples?

File: src/wrapper/clap-juce-wrapper.cpp [hunk 2]
```diff
@@ -197,17 +199,36 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
 
     void audioProcessorChanged(juce::AudioProcessor *proc, const ChangeDetails &details) override
     {
-        juce::ignoreUnused(proc, details);
-        DBG("audio processor changed");
+        juce::ignoreUnused(proc);
         if (details.latencyChanged)
-            DBG("   Unsupported currently - latencyChanged");
+        {
+            runOnMainThread([this] {
```
Comment (jatinchowdhury18) [inline src/wrapper/clap-juce-wrapper.cpp:205]: I wonder if the other changes should be forced to run on the main thread also?

**Comment by baconpaul** [comment]
Thank you!

so here’s the problem 

https://github.com/free-audio/interop-tracker/issues/8

that issue means if we turn this on then bitwig restarts surge every time you change an oscillator type because it over reacts to these messages - Id been holding off on this change until that was fixed. Did you try surge in beta2 with this? If not I can today and see if beta2 has fixed it

**Comment by baconpaul** [comment]
Also yes everything has to run on the main thread otherwise bitwig will kill your plugin. (My original version before I made it a debug was a juce::timer::runlater of host->blah

**Comment by jatinchowdhury18** [comment]
Gotcha! I've only tested these changes with BYOD so far. I can give Surge a try this evening with beta2.

**Comment by baconpaul** [comment]
I just gave it a  try (after moving everything to main thread - which I pushed back up to your branch)

When I change an oscillator type in surge (which triggers the paramInfoChanged callback) bitwig restarts processing.

```
Restarting processing after parameters updated
Applying plugin IO state
PluginHost: Requesting restart of audio processing
Requesting restart of processing because IO buses or latency have changed
PluginDeviceContentsNode request restart processing
Starting fade out because request restart processing
```

I'll merge this anyway since it is correct and will also pull surge up to use it but we really need a fix for this before bitwig comes out of beta!

**Pull request merged**